src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -202,6 +229,13 @@ type CriteriaGroupViewProps = {
 
 export const CriteriaGroupView: React.FC<CriteriaGroupViewProps> = (props) => {
   const { index, criteriaGroup, updateCohort, cohort, datasetDetails } = props;
+
+  const deleteCriteria = (criteria: AnyCriteria) =>
+    updateCohort(_.set(`criteriaGroups.${index}.criteria`, _.without([criteria], criteriaGroup.criteria)));
+
+  const updateCriteria = (criteria: AnyCriteria, criteriaIndex: number) =>
+    updateCohort(_.set(`criteriaGroups.${index}.criteria.${criteriaIndex}`, criteria));
```

Comment:
Since `Criteria` has `id` you could use that to update it instead of `index`. Not sure if using one is better than the other though.

Reply:
If you did use `Criteria.id` instead, you could change the code back to use `createCriteriaViewComponent()` with two arguments instead of one.

Reply:
Just double checking from a UX standpoint: the list of criteria is static?  E.g. I wonder if we envision a case where the criteria might be refetched (which might cause indexes to get out of sync)

Reply:
Yes, the list of criteria is static!